### PR TITLE
Extend i64 integer arithmetic vm tests

### PR DESCRIPTION
**Pull request by marbre**
Takes over the i32 tests to test the i64 integer arithmetic ops.

### CODE REVIEW

File: iree/vm/test/arithmetic_ops_i64.mlir [hunk 1]
```diff
@@ -1,10 +1,22 @@
 vm.module @arithmetic_ops_i64 {
 
   //===--------------------------------------------------------------------===//
-  // I64 Arithmetic
+  // ExtI64: Native integer arithmetic
   //===--------------------------------------------------------------------===//
 
+  // TODO: The CModuleTarget enforces exports to be ordered.
```
Comment (benvanik) [inline iree/vm/test/arithmetic_ops_i64.mlir:7]: can we have a tracking issue for this to chat about it?
(for example we may be able to add a pass you could run to reorder if you need them to be in a certain order)

**Review by benvanik — APPROVED** [review]

File: iree/vm/test/arithmetic_ops_i64.mlir (hunk 1 above)
Comment (marbre) [reply iree/vm/test/arithmetic_ops_i64.mlir:7]: We have an internal issue for what we have left to do regarding EmitC. And yeah, as the bytecode module uses `llvm::sort` 
https://github.com/google/iree/blob/96cdbe372cc01a05408a39f1a86e17be51c09ffb/iree/compiler/Dialect/VM/Target/Bytecode/BytecodeModuleTarget.cpp#L92-L101
So we want to make sure the C module behaves similar to the bytecode module.

File: iree/vm/test/arithmetic_ops_i64.mlir [hunk 2]
```diff
@@ -1,10 +1,22 @@
...
+  vm.export @test_mul_i64
+  vm.export @test_not_i64
+  vm.export @test_or_i64
+  vm.export @test_rem_i64s
+  vm.export @test_rem_i64u
+  vm.export @test_sub_i64
+  vm.export @test_xor_i64
+
   vm.func @test_add_i64() {
     %c1 = vm.const.i64 1 : i64
```
Comment (simon-camp) [inline iree/vm/test/arithmetic_ops_i64.mlir:21]: How about using values, which would overflow for 32bit integers?

Comment (marbre) [reply iree/vm/test/arithmetic_ops_i64.mlir:21]: Yeah, I had this in mind as well... But, hm, `test_add_i64` was already present ;)

**Pull request merged**